test/streams.js

```diff
@@ -0,0 +1,36 @@
+var nano = require('..');
+var test = require('tape');
+
+test('pipe a thousand msgs between incompatible socket types', function(t){
+
+  var sent = 0, recv = 0;
+
+  var pub = nano.socket('pub', { tcpnodelay: true });
+  var sub = nano.socket('sub', { tcpnodelay: true });
+
+  var push = nano.socket('push');
+  var pull = nano.socket('pull');
+  pull.setEncoding('utf8'); //should also be able to do in `socket(type, opts)`
```

Comment:
stream default is a buffer, yes.

With node streams, we have a different way of asking the lib to return back a string. before everything node-nanomsg returned was a buffer